src/LBQuoter.sol

```diff
@@ -147,18 +171,55 @@ contract LBQuoter {
         quote.virtualAmountsWithoutSlippage[0] = amountIn;
 
         for (uint256 i; i < swapLength; i++) {
-            if (_factoryV2 != address(0)) {
+            if (_factoryV2_2 != address(0)) {
+                // Fetch swaps for V2.2
+                ILBFactory.LBPairInformation[] memory LBPairsAvailable =
+                    ILBFactory(_factoryV2_2).getAllLBPairs(IERC20(route[i]), IERC20(route[i + 1]));
+
+                if (LBPairsAvailable.length > 0 && quote.amounts[i] > 0) {
+                    for (uint256 j; j < LBPairsAvailable.length; j++) {
+                        if (!LBPairsAvailable[j].ignoredForRouting) {
+                            bool swapForY = address(LBPairsAvailable[j].LBPair.getTokenY()) == route[i + 1];
+
+                            try ILBRouter(_routerV2_2).getSwapOut(
+                                LBPairsAvailable[j].LBPair, quote.amounts[i], swapForY
+                            ) returns (uint128 amountInLeft, uint128 swapAmountOut, uint128 fees) {
+                                if (amountInLeft == 0 && swapAmountOut > quote.amounts[i + 1]) {
+                                    quote.amounts[i + 1] = swapAmountOut;
+                                    quote.pairs[i] = address(LBPairsAvailable[j].LBPair);
+                                    quote.binSteps[i] = uint16(LBPairsAvailable[j].binStep);
+                                    quote.versions[i] = ILBRouter.Version.V2_2;
+
+                                    // Getting current price
+                                    uint24 activeId = LBPairsAvailable[j].LBPair.getActiveId();
+                                    quote.virtualAmountsWithoutSlippage[i + 1] = _getV2Quote(
+                                        quote.virtualAmountsWithoutSlippage[i] - fees,
+                                        activeId,
+                                        quote.binSteps[i],
+                                        swapForY
+                                    );
+
+                                    quote.fees[i] = ((uint256(fees) * 1e18) / quote.amounts[i]).safe128(); // fee percentage in amountIn
+                                }
+                            } catch {}
+                        }
+                    }
+                }
+            }
```

Comment:
This code is the same for 2.1 and 2.2 right ?

Reply:
Yes, the ABIs are the same for this reason

Reply:
It could be abstracted better in a separate functions, but I don't think it's worth changing the quoter